test/functional/test_framework/test_framework.py

```diff
@@ -150,31 +149,28 @@ def main(self):
         else:
             self.log.warning("Not cleaning up dir %s" % self.options.tmpdir)
             if os.getenv("PYTHON_DEBUG", ""):
-                # Dump the end of the debug logs, to aid in debugging rare
-                # travis failures.
-                import glob
-                filenames = [self.options.tmpdir + "/test_framework.log"]
-                filenames += glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
-                MAX_LINES_TO_PRINT = 1000
-                for fn in filenames:
-                    try:
-                        with open(fn, 'r') as f:
-                            print("From", fn, ":")
-                            print("".join(deque(f, MAX_LINES_TO_PRINT)))
-                    except OSError:
-                        print("Opening file %s failed." % fn)
-                        traceback.print_exc()
+                logging.shutdown()  # Flush handlers
```

Comment:
Python reference: https://docs.python.org/3/library/logging.html#logging.shutdown :

> This should be called at application exit and no further use of the logging system should be made after this call.

You're calling `log.info()` after this call, and also calling `logging.shutdown()` a second time.

Perhaps to avoid this, you could move this whole block to below the last call to `self.log`?

Reply:
Should be fine. I call `sys.exit` a few lines below.